mypy/checker.py

```diff
@@ -3648,10 +3642,12 @@ def is_more_precise_or_partially_overlapping(t: Type, s: Type) -> bool:
     return (is_callable_compatible(signature, other,
                                    is_compat=is_more_precise_or_partially_overlapping,
                                    is_compat_return=lambda l, r: not is_subtype(l, r),
-                                   check_args_covariantly=True) or
+                                   check_args_covariantly=True,
+                                   allow_potential_compatibility=True) or
```

Comment:
TBH, I don't like the term "potential compatibility". Why not "partial overlap"? After all, IIUC, this is the same as `Union[A, B]` vs `Union[B, C]` but for arg counts.